egui/src/widgets/plot/items/mod.rs

```diff
@@ -1575,44 +1594,41 @@ fn add_rulers_and_text(
     plot: &PlotConfig<'_>,
     text: Option<String>,
     shapes: &mut Vec<Shape>,
+    cursors: &mut Vec<Cursor>,
 ) {
     let orientation = elem.orientation();
     let show_argument = plot.show_x && orientation == Orientation::Vertical
         || plot.show_y && orientation == Orientation::Horizontal;
     let show_values = plot.show_y && orientation == Orientation::Vertical
         || plot.show_x && orientation == Orientation::Horizontal;
 
-    let line_color = rulers_color(plot.ui);
-
     // Rulers for argument (usually vertical)
     if show_argument {
-        let push_argument_ruler = |argument: PlotPoint, shapes: &mut Vec<Shape>| {
-            let position = plot.transform.position_from_point(&argument);
-            let line = match orientation {
-                Orientation::Horizontal => horizontal_line(position, plot.transform, line_color),
-                Orientation::Vertical => vertical_line(position, plot.transform, line_color),
+        let push_argument_ruler = |argument: PlotPoint, cursors: &mut Vec<Cursor>| {
+            let cursor = match orientation {
+                Orientation::Horizontal => Cursor::horizontal(argument.y),
+                Orientation::Vertical => Cursor::vertical(argument.x),
             };
-            shapes.push(line);
+            cursors.push(cursor);
         };
 
         for pos in elem.arguments_with_ruler() {
-            push_argument_ruler(pos, shapes);
+            push_argument_ruler(pos, cursors);
```

Comment:
Not new in this PR, but I really don't see why `push_argument_ruler` exists and is not just inlined in the loop